test/Microsoft.Azure.Devices.Edge.Test/PriorityQueues.cs

```diff
@@ -124,32 +128,159 @@ public async Task PriorityQueueModuleToModuleMessages()
+            await this.PollUntilFinishedAsync(relayerModuleName, token);
+
+            HttpClient client = new HttpClient();
+            HttpResponseMessage response = await client.GetAsync("http://localhost:5001/api/report");
+            var jsonstring = await response.Content.ReadAsStringAsync();
+            bool isPassed = (bool)JArray.Parse(jsonstring)[0]["IsPassed"];
+            if (!isPassed)
+            {
+                Log.Verbose("Test Result Coordinator response: {Response}", jsonstring);
+            }
+
+            Assert.IsTrue(isPassed);
+        }
+
+        private Dictionary<string, object> BuildRoutes(string[] priorities, string[] ttls, string sendModule, string receiveModule)
         {
             Dictionary<string, object> routes = new Dictionary<string, object>();
-            foreach (string priority in priorities)
+            for (int i = 0; i < priorities.Length; i++)
             {
+                string ttl = ttls[i];
+                string priority = priorities[i];
                 // If we encounter "Default" in the priority list, don't add a priority - the default priority will automatically get picked up
-                if (priority.Contains(TestConstants.PriorityQueues.Default))
+                var routeInfo = new Dictionary<string, object>();
+                routeInfo["route"] = $"FROM /messages/modules/{sendModule}/outputs/pri{priority} INTO BrokeredEndpoint('/modules/{receiveModule}/inputs/input1')";
+                if (!priority.Contains(TestConstants.PriorityQueues.Default))
                 {
-                    routes.Add($"LoadGenToRelayer{priority}", new Dictionary<string, object>
-                    {
-                        ["route"] = $"FROM /messages/modules/{sendModule}/outputs/pri{priority} INTO BrokeredEndpoint('/modules/{receiveModule}/inputs/input1')",
-                    });
+                    routeInfo["priority"] = int.Parse(priority);
                 }
-                else
+
+                if (int.Parse(ttl) != -1)
                 {
-                    routes.Add($"LoadGenToRelayer{priority}", new Dictionary<string, object>
-                    {
-                        ["route"] = $"FROM /messages/modules/{sendModule}/outputs/pri{priority} INTO BrokeredEndpoint('/modules/{receiveModule}/inputs/input1')",
-                        ["priority"] = int.Parse(priority)
-                    });
+                    routeInfo["timeToLiveSecs"] = int.Parse(ttl);
                 }
+
+                routes.Add($"LoadGenToRelayer{priority}", routeInfo);
             }
 
             return routes;
         }
 
+        private string BuildTTLString(int numberOfTTLs)
+        {
+            Random rng = new Random();
+            // Choose from a set of 2 TTLs below the threshold, 2 above
+            var ttlSet = new string[] { "5", "10", "400", "1600" };
```

Comment:
How long will this test take to complete (on average)?

Reply:
~2-2.5 minutes on amd64